contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -504,45 +495,54 @@ library SupplyLogic {
         mapping(uint256 => address) storage reservesList,
         DataTypes.UserConfigurationMap storage userConfig,
         address asset,
-        uint256 tokenId,
+        uint256[] calldata tokenIds,
         bool useAsCollateral,
         uint256 reservesCount,
         address priceOracle
     ) external {
         DataTypes.ReserveData storage reserve = reservesData[asset];
         DataTypes.ReserveCache memory reserveCache = reserve.cache();
 
-        (
-            bool valid,
-            address owner,
-            uint256 collaterizedBalance
-        ) = ICollaterizableERC721(reserveCache.xTokenAddress)
-                .setIsUsedAsCollateral(tokenId, useAsCollateral);
+        bool isAnyValid = false;
+        address sender = msg.sender;
+        for (uint256 index = 0; index < tokenIds.length; index++) {
+            bool valid = ICollaterizableERC721(reserveCache.xTokenAddress)
+                .setIsUsedAsCollateral(
```

Comment:
we should wrap batch operation inside NToken

Reply:
yeah. we can add a function to support batch operation.